### PR TITLE
Avoid escaping paths when editing credentials

### LINKED ISSUE #41617: change_credentials_in_system_editor breaks with spaces in Windows user name
### Steps to reproduce

Run `rails credentials:edit` while using a windows username that contains spaces. In my case, it's "Lucius Riccio" (since this was my dad's laptop a decade ago)

### Expected behavior

Open a temporary file with a path like `C:/Users/Lucius Riccio/AppData/Local/Temp/22716.credentials.yml`
### Actual behavior

Opens two temporary files with vscode, like `Lucius` and a file named  Riccio/AppData/Local/Temp/22716.credentials.yml.

### System configuration
**Rails version**:
6.1.2.1
**Ruby version**:
3.0.0

[`change_credentials_in_system_editor`](https://github.com/rails/rails/blob/fdba9304c2652af674fdb9cf3704ff5a64af92d7/railties/lib/rails/commands/credentials/credentials_command.rb#L92) should handle paths with spaces.

### PR DESCRIPTION
**Pull request by jonathanhefner**
`Shellwords.escape` escapes unquoted spaces with a backslash, but Windows does not treat backslash as an escape character.  Escaping is also a problem when paths are expressed in shortened 8.3 format (e.g. `C:\Users\RubyOn~1\AppData\Local\Temp\...`) because a backslash will be erroneously added before the `~`.

We can avoid the need to escape by using `system(command_name, *args)` instead of `system(command_line)`, but we must still support `ENV["EDITOR"]` values that embed command line arguments, such as `subl -w`.

This commit changes to `system(command_name, *args)`, but uses `Shellwords.split` to extract any embedded arguments from `ENV["EDITOR"]`.  This requires that Windows users put quotes around the entire path of their editor if it contains spaces, such as:

```
SET EDITOR="C:\Program Files\Microsoft VS Code\Code.exe" -w
```

In other words, the following are **not** supported on Windows:

```
SET "EDITOR=C:\Program Files\Microsoft VS Code\Code.exe"
SET EDITOR=C:\Program Files\Microsoft VS Code\Code.exe
SET EDITOR=C:\"Program Files"\"Microsoft VS Code"\Code.exe -w
SET EDITOR=C:\Program^ Files\Microsoft^ VS^ Code\Code.exe -w
SET EDITOR=C:\Program` Files\Microsoft` VS` Code\Code.exe -w
```

Fixes #41617 (again).
Closes #44890.

---

This is based on my suggestion in https://github.com/rails/rails/pull/42728#discussion_r666287484, with a fix for the test I mentioned in https://github.com/rails/rails/pull/42728#issuecomment-876585266.

/cc @ariccio @joshleblanc Would it be possible for you to test this change locally, and verify that it solves your issue?

### CODE REVIEW

**Comment by joshleblanc** [comment]
This change works on my end, and is a better solution than mine.

FWIW, your assumption that the temp path was coming from the environment variables was correct. It's set with the 8.3 filename in both TMP and TMP.
